modules/validations/connected_ways.js

```diff
@@ -0,0 +1,428 @@
+                segment2Info = segmentInfos[j];
+
+                // don't check for self-intersection in this validation
+                if (segment2Info.wayId === way1.id) continue;
+
+                // skip if this way was already checked and only one issue is needed
+                if (checkedSingleCrossingWays[segment2Info.wayId]) continue;
+
+                // mark this way as checked even if there are no crossings
+                comparedWays[segment2Info.wayId] = true;
+
+                way2 = graph.hasEntity(segment2Info.wayId);
+                if (!way2) continue;
+                taggedFeature2 = getFeatureWithFeatureTypeTagsForWay(way2, graph);
+                way2FeatureType = getFeatureType(taggedFeature2, graph);
+
+                if (way2FeatureType === null ||
+                    isLegitCrossing(taggedFeature1.tags, way1FeatureType, taggedFeature2.tags, way2FeatureType)) {
+                    continue;
+                }
+
+                // create only one issue for building crossings
+                oneOnly = way1FeatureType === 'building' || way2FeatureType === 'building';
+
+                nAId = segment2Info.nodes[0];
+                nBId = segment2Info.nodes[1];
+                intersectingNode = null;
+                const hasConnectionNode = (nAId === n1.id || nAId === n2.id || nBId === n1.id || nBId === n2.id);
+                if (!hasConnectionNode) {
+                        continue;
+                }
+                // check tags of intersecting node
+                intersectingNode = [nAId, nBId].find(id => id === n1.id || id === n2.id);
+                let intersectingNodeEntity = graph.entity(intersectingNode);
+                if (way1FeatureType === 'power' || way2FeatureType === 'power') {
+                    if (['insulator', 'terminal'].includes(intersectingNodeEntity.tags.power)) {
+                        continue;
+                    }
+                }
+
+                nA = graph.hasEntity(nAId);
+                if (!nA) continue;
+                nB = graph.hasEntity(nBId);
+                if (!nB) continue;
+
+                segment1 = [n1.loc, n2.loc];
+                segment2 = [nA.loc, nB.loc];
+                var point = geoLineIntersection(segment1, segment2);
+                if (point) {
+                    ConnectionsInfo.push({
+                        wayInfos: [
+                            {
+                                way: way1,
+                                featureType: way1FeatureType,
+                                edge: [n1.id, n2.id]
+                            },
+                            {
+                                way: way2,
+                                featureType: way2FeatureType,
+                                edge: [nA.id, nB.id]
+                            }
+                        ],
+                        crossPoint: point,
+                        crossIndex: way1NodesId.indexOf(intersectingNode),
```

Comment:
Since this is only called once per issue, I don't think it's worth it to precompute `way1NodesId`.
```suggestion
                        crossIndex: way1Nodes.findIndex(n => n.id === intersectingNode),
```

Reply:
This creates unsafe reference to the intersectingNode variable

Reply:
Not if you declare `intersectingNode` lazily in the loop when only the intersections remain:
```js
                const intersectingNode = [nAId, nBId].find(id => id === n1.id || id === n2.id);

```

Reply:
ok, it makes sense, thanks !